CMSgov/eAPD
# Assurances and Compliance error messages

Resolves #3358 

### Description
Adds error messages to assurances and compliance. If yes or no isn't selected, an error message of "Select yes or no" is displayed under the radio button. If no is selected but there is no explanation, then "Provide an explanation" is displayed under the text box.

### Automated test cases written
Updated the unit test to use react-testing-library instead of enzyme.

| Test Type | Given | When | Then |
|---|---|----|---|
| unit | adminCheck is off | page loads | no errors are displayed |
| unit | adminCheck is on | page loads | displays "Select yes or no" for all of unanswered radio buttons |
| unit | adminCheck is on | yes is selected for the radio button | the radio error goes away |
| unit | adminCheck is on | no is selected | the radio button error goes away, the explanation text area is displayed, and "Provide an explanation" is displayed under the text area |
| unit | adminCheck is on | no was selected and then yes is selected again | the radio error goes away |
| unit | adminCheck is on | no is selected and an explanation is added | the radio error and the text area error go away |
| e2e | an empty APD | the page loads | displays "Select yes or no" for all 15 radio buttons |
| e2e | an empty APD | the user selects values for all 15 radio buttons | all of the error messages go away

### Steps to manually verify this change

1. create an APD
2. navigate to Assurances and Compliance
3. confirm that all of the radio buttons display error message
4. select yes on one of the radio buttons
5. confirm that the error message goes away
6. select no on one of the radio buttons
7. confirm that the text area opens up and displays an error message
8. select yes on the same radio button
9. confirm that the error message goes away again
10. select no on the same radio button again
11. enter text in the text area that opens
12. confirm that the text area error message goes away

### This pull request is ready to review when


### This pull request can be merged when

- [x] QA has verified the accessibility using tools such as screen readers and WAVE (accessibility evaluation tool)
